iocore/net/I_NetVConnection.h

```diff
@@ -187,6 +187,8 @@ struct NetVCOptions {
   uint32_t packet_tos;
   uint32_t packet_notsent_lowat;
 
+  uint32_t http2_buffer_water_mark;
```

Comment:
I'm ok with exposing a setting tunable via sni.yaml and I'm not going to block this PR because we can change the internal implementation later without breaking compatibility, but I don't like to have this option here because it's for a specific application protocol (not for generic connections).

I'd put it like:

```
class TLSSNISupport {
  ...
  struct HintsFromSNI { // NEW
    uint32_t http2_buffer_water_mark;
  } hints_from_sni;
};
```

It'd still have an application specific setting in one of TLS stuff, but it's better than contaminating NetVCOptions, IMO. And it also suggests the value is just a hint, and you can ignore it if appropriate (remember connection coalescing?).

Reply:
That sounds reasonable. 

And yes, I remember connection coalescing. I work for a CDN that tends to keep SANs for certs separated by SNI. ;) How this works today, is perfect for our use case. But I understand others probably do not work in the same way.